### PR TITLE
Rename all shadowed types and variables and enable Wshadow when in pedantic mode

### PR DESCRIPTION
**Pull request by jgopel**
I agree that my contributions are licensed under the {fmt} license, and agree to future changes to the licensing.

### CODE REVIEW

File: include/fmt/os.h [hunk 1]
```diff
@@ -363,8 +363,8 @@ struct ostream_params {
   ostream_params() {}
 
   template <typename... T>
-  ostream_params(T... params, int oflag) : ostream_params(params...) {
-    this->oflag = oflag;
+  ostream_params(T... params, int new_oflag) : ostream_params(params...) {
+    this->oflag = new_oflag;
```
Comment (vitaut) [inline include/fmt/os.h:367]: Please remove `this->`.

File: test/core-test.cc [hunk 2]
```diff
@@ -213,12 +213,12 @@ struct custom_context {
 
   template <typename T> struct formatter_type {
     template <typename ParseContext>
-    auto parse(ParseContext& ctx) -> decltype(ctx.begin()) {
-      return ctx.begin();
+    auto parse(ParseContext& parse_ctx) -> decltype(parse_ctx.begin()) {
```
Comment (vitaut) [inline test/core-test.cc:216]: Let's keep argument names as is but rename the data member to `parse_ctx` since it has wider scope.

File: test/format-test.cc [hunk 3]
```diff
@@ -219,12 +219,12 @@ TEST(AllocatorTest, allocator_ref) {
   check_forwarding(alloc, ref3);
 }
 
-typedef allocator_ref<std::allocator<char>> TestAllocator;
+typedef allocator_ref<std::allocator<char>> StdAllocator;
```
Comment (vitaut) [inline test/format-test.cc:222]: Let's follow the naming conventions while at it:

```c++
using std_allocator = allocator_ref<std::allocator<char>>;
```

**Review by vitaut — CHANGES_REQUESTED** [review]
Thanks for the PR. Mostly looks good but please fix CI build failures and address inline comments.

File: include/fmt/os.h (hunk 1 above)
Comment (jgopel) [reply include/fmt/os.h]: Done

File: test/core-test.cc (hunk 2 above)
Comment (jgopel) [reply test/core-test.cc]: Done - I'm glad you looked at that one, I had a feeling that the situation might be something like this but I didn't want to be too aggressive so I tried my best to keep the scope very local.

File: test/format-test.cc (hunk 3 above)
Comment (jgopel) [reply test/format-test.cc]: 👍 Done

**Comment by vitaut** [comment]
A few more build failures: https://travis-ci.org/github/fmtlib/fmt/jobs/740386845

**Comment by jgopel** [comment]
I can't get this last set of errors to reproduce locally for reasons that are not clear to me, so there might be a bit of churn with using CI to find the errors.

**Comment by jgopel** [comment]
Just in case anyone else encounters the issue of not being able to reproduce errors locally - the C++ standard seems to matter to at least GCC for whether or not ` -Wshadow` generates warnings in certain narrow contexts.

File: test/format-test.cc [hunk 4]
```diff
@@ -105,7 +105,7 @@ struct uint32_pair {
 TEST(UtilTest, BitCast) {
   auto s = fmt::detail::bit_cast<uint32_pair>(uint64_t{42});
   EXPECT_EQ(fmt::detail::bit_cast<uint64_t>(s), 42ull);
-  s = fmt::detail::bit_cast<uint32_pair>(uint64_t(~0ull));
+  s = fmt::detail::bit_cast<uint32_pair>(~0ull);
```
Comment (vitaut) [inline test/format-test.cc:108]: This looks unrelated and technically not correct.

File: test/std-format-test.cc [hunk 5]
```diff
@@ -102,7 +102,7 @@ template <> struct std::formatter<S> {
   size_t width_arg_id = 0;
 
   // Parses a width argument id in the format { <digit> }.
-  constexpr auto parse(format_parse_context& ctx) {
+  auto parse(format_parse_context& ctx) {
```
Comment (vitaut) [inline test/std-format-test.cc:105]: Please revert.

**Review by vitaut — CHANGES_REQUESTED** [review]

File: test/std-format-test.cc (hunk 5 above)
Comment (jgopel) [reply test/std-format-test.cc]: Whoops, sorry about that - fixed

File: test/format-test.cc (hunk 4 above)
Comment (jgopel) [reply test/format-test.cc]: Whoops, sorry about that - fixed.

**Comment by darklukee** [comment]
@jgopel thank you for doing this. It was annoying my team for some time now :)

**Comment by vitaut** [comment]
Merged, thanks!
